### ISSUE TITLE
[Globalstep] When user taps multiple times on "+" icon to add  multiple buttons, the whole screen goes haywired.

### ISSUE THREAD

**Issue by gglobalstep** [wordpress-mobile/gutenberg-mobile#2238]
Environment:
iOS: iPad Air 2(v13.3.1), iPad (6th Gen) - v12.4

Steps To Reproduce:
 
 1. Install and launch the application.
 2. Tap on "Log In".
 3. Log in to the app using valid credentials.
 4. Tap on "My site" and create a site.
 5.  Tap on Site Pages.
 6.  Tap on 'Drafts and then on +' icon to create a site page.
 7. Tap on "+" blue icon below near the keyboard.
 8.  Tap on "Buttons" icon.
 9. Tap multiple times on side "+" icon to add multiple buttons.
10. Observe that when you tap multiple times on side "+"icon to add buttons, the whole screen goes haywired. 
  
 Actual Result:
When user taps multiple times on side "+" icon to add multiple buttons, the whole screen goes haywired.
 
 Expected Result:
Application should add multiple buttons,when user taps on "+" side icon.

[#2238.zip](https://github.com/wordpress-mobile/gutenberg-mobile/files/4594786/2238.zip)

**Comment by SergioEstevao** [comment]
@pinarol was this something that happened on previous versions? I'm unable to reproduce on the current version in develop.

**Comment by pinarol** [comment]
Not that I know of. I couldn't repro this on 1.28 release branch. The attachment video seems corrupted, @gglobalstep could you upload the video again?

@lukewalczak any idea if this happened on previous versions?

**Comment by pinarol** [comment]
@gglobalstep could you also add which WordPress version was used when testing?

**Comment by lukewalczak** [comment]
Didn't face something like that before and cannot reproduce it right now as well (testing on iPad simulator). I would like to see the gif/video, maybe it will say something more about weird behavior.

**Comment by designsimply** [comment]
I was also unable to replicate the problem where tapping the inline appender in the buttons block results in the screen going haywire. ([33s](https://cloudup.com/coBmETSvbzy))

<sup>Tested with WPiOS 14.8.0.2 TestFlight beta on iPad Mini 4 (WiFi) (iPad5,1) iPadOS 13.4.1 on a WordPress.com Free site (shown in video) and I also tested from a Jetpack and Self-hosted site as well.</sup>

I noticed the editor toolbar sometimes flies down then back up again or the keyboard blinks out and back again once when the inline appender is tapped which feels a little bouncy but isn't anything like a focus loop we were seeing in some other recently reported cases for other blocks. Note: the keyboard getting dismissed and re-opened a few times, as seen in the video linked above, were all due to mis-tapping the appender in my case.

**Comment by gglobalstep** [comment]
This issue is observed to be "Open" on following environment:
iOS: iPhone 8 (v13.4.1), iPhone 11 Pro (v13.2.2), iPhone XS (v13.2.2).
Build [version]: v14.9

Attachment:
[#2238.zip](https://github.com/wordpress-mobile/gutenberg-mobile/files/4663371/2238.zip)

**Comment by pinarol** [comment]
@gglobalstep 👋 The video only provides the focus loop, I can't see the steps that cause this behavior. Could you send that as well? I am double tapping the side "+" icon and can't repro this. Are you maybe tapping too fast too many times? 

Currently the editor has some known vulnerabilities about focus loops if you do some super quick actions sequentially on rich text based blocks. We are planning to do a refactoring about this but we need React Native's v0.63.0 to do that. (Here's some [technical analysis](https://github.com/wordpress-mobile/gutenberg-mobile/pull/1702#issuecomment-567831623) about this.) 

If this is not likely to happen with a usual user behavior I'd say let's try to fix this when we upgrade to React Native v0.63.0 unless the steps show that it is likely to be produced in normal usage as well.

**Comment by antonis** [comment]
I was able to reproduce the reported issue on an [iPhone SE 2020 with iOS 14 ](https://cloudup.com/c5I-Ln80Paf) and an iPhone 11 simulator (iOS 13.6) running WordPress iOS or even [the demo app](https://cloudup.com/cVkSn9NCfLt).

To trigger this behaviour the **+** button should be pressed really fast a lot of times and not with usual user behavior.

I tried the same steps on Android but it only resulted a in [temporary freeze and the buttons are eventually created](https://cloudup.com/cnj85DJY3cC).
